cgen error when spawn anon functions, with closure parameters and mut ref function parameters

### Describe the bug

```v
import net

fn main() {
	mut conn := &net.TcpConn{}
	a := 1

	spawn fn [a] (mut conn net.TcpConn) {
	}(mut conn)
}
```

### Reproduction Steps

```
/tmp/v_501/a.2665062129621293191.tmp.c:2158:50: error: expected identifier or '('
typedef struct thread_arg___v_void_net__TcpConn* {
                                                 ^
/tmp/v_501/a.2665062129621293191.tmp.c:2161:3: error: must use 'struct' tag to refer to type 'thread_arg___v_void_net__TcpConn'
} thread_arg___v_void_net__TcpConn*;
  ^
  struct 
/tmp/v_501/a.2665062129621293191.tmp.c:2161:36: error: expected identifier or '('
} thread_arg___v_void_net__TcpConn*;
                                   ^
/tmp/v_501/a.2665062129621293191.tmp.c:2162:29: error: expected ';' after top level declarator
 void* __v_void_net__TcpConn*_thread_wrapper(thread_arg___v_void_net__TcpConn* *arg);
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Expected Behavior

compile passed.

### Current Behavior

error.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

latest master

### Environment details (OS name and version, etc.)

macos, centos8
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/utils.v
+++ b/vlib/v/gen/c/utils.v
@@ -113,9 +113,9 @@ fn (mut g Gen) fn_var_signature(return_type ast.Type, arg_types []ast.Type, var_
 	return sig
 }
 
-// generate anon fn cname, e.g. `anon_fn_void_int_string`
+// generate anon fn cname, e.g. `anon_fn_void_int_string`, `anon_fn_void_int_ptr_string`
 fn (mut g Gen) anon_fn_cname(return_type ast.Type, arg_types []ast.Type) string {
-	ret_styp := c_name(g.typ(return_type))
+	ret_styp := c_name(g.typ(return_type).replace('*', '_ptr'))
 	mut sig := '${ret_styp}_'
 	for j, arg_typ in arg_types {
 		arg_sym := g.table.sym(arg_typ)
@@ -124,7 +124,7 @@ fn (mut g Gen) anon_fn_cname(return_type ast.Type, arg_types []ast.Type) string 
 			arg_sig := g.anon_fn_cname(func.return_type, func.params.map(it.typ))
 			sig += 'fnon_fn_' + arg_sig
 		} else {
-			arg_styp := c_name(g.typ(arg_typ))
+			arg_styp := c_name(g.typ(arg_typ).replace('*', '_ptr'))
 			sig += arg_styp
 		}
 		if j < arg_types.len - 1 {

--- a/vlib/v/tests/anon_fn_test.v
+++ b/vlib/v/tests/anon_fn_test.v
@@ -41,3 +41,16 @@ fn test_anon_fn_returning_a_mut_parameter_should_act_the_same_as_normal_fn_retur
 	y := fnormal(mut b, 123)
 	assert a == b
 }
+
+// for issue 20163
+struct Struct {}
+
+fn test_spawn_anon_fn_with_closure_parameters_and_mut_ref_parameters() {
+	mut s := &Struct{}
+	a := 1
+
+	t := spawn fn [a] (mut s Struct) int {
+		return a
+	}(mut s)
+	assert t.wait() == 1
+}

